compose-app/src/main/java/com/mapbox/maps/extension/compose/MapApplier.kt

```diff
@@ -0,0 +1,47 @@
+package com.mapbox.maps.extension.compose
+
+
+import androidx.compose.runtime.AbstractApplier
+import com.mapbox.maps.MapControllable
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+
+internal interface MapNode {
+  fun onAttached() {}
+  fun onRemoved() {}
+  fun onCleared() {}
+}
+
+private object MapNodeRoot : MapNode
+
+internal class MapApplier(
```

Comment:
some docs would be nice but not important now.

Reply:
this is just a prototyping branch, production PRs will be cut separately.